### PR TITLE
Make Name a newtype

### PR DESCRIPTION
**Pull request by Ericson2314**
The GLR backend has me stuck, but I think I got the rest.

Depends on #296

### CODE REVIEW

**Comment by Ericson2314** [comment]
I think I am getting somewhere wit this now

**Comment by Ericson2314** [comment]
It builds, but the tests aren't passing. Broke off #296 in part to help bisect.

File: lib/tabular/src/Happy/Tabular/NameSet.hs [hunk 1]
```diff
@@ -1,8 +1,84 @@
...
+--
+
+union :: NameSet -> NameSet -> NameSet
+union = coerce IntSet.union
+
+unions :: (Functor f, Foldable f) => f NameSet -> NameSet
+unions = coerce . IntSet.unions . fmap coerce
+
+difference :: NameSet -> NameSet -> NameSet
+difference = coerce difference
```
Comment (sgraf812) [inline lib/tabular/src/Happy/Tabular/NameSet.hs:71]:
```suggestion
difference :: NameSet -> NameSet -> NameSet
difference = coerce IntSet.difference
```

Here you go! Just took me an hour of debugging (once you start investigating ...)

File: lib/tabular/src/Happy/Tabular/NameSet.hs [hunk 2]
```diff
@@ -1,8 +1,84 @@
...
+
+(\\) :: NameSet -> NameSet -> NameSet
+(\\) = coerce (\\)
```
Comment (sgraf812) [inline lib/tabular/src/Happy/Tabular/NameSet.hs:74]:
```suggestion
(\\) = coerce (IntSet.\\)
```

File: lib/tabular/src/Happy/Tabular/NameSet.hs (hunk 1 above)
Comment (Ericson2314) [reply lib/tabular/src/Happy/Tabular/NameSet.hs:71]: Wow! Thank you!!!!

**Comment by Ericson2314** [comment]
OK @sgraf812!

**Review by sgraf812 — APPROVED** [review]
Thanks!

**Pull request merged**